modules/common/src/bugsnag/bugsnag.module.ts

```diff
@@ -0,0 +1,31 @@
+import Bugsnag, {Config} from '@bugsnag/js';
+import {DynamicModule, Module} from '@nestjs/common';
+import {BugsnagClient} from './bugsnag.client';
+
+@Module({
+    providers: [
+        {
+            provide: BugsnagClient,
+            useFactory: async (): Promise<BugsnagClient> => BugsnagModule.providedClient,
+        },
+    ],
```

Comment:
Why you need a provide/factory here and below?

Reply:
For in `forRoot` I need a factory so that I can pass in the Bugsnag config. Here, I need it because otherwise this provider is always `undefined` when it arrives to `Logger`.. 

It's essentially the same solution that we use(d) in our database module, but there we provide `DatabaseProvider`